modular_sand/code/datums/interactions/lewd_definitions.dm

```diff
@@ -761,15 +772,27 @@
 			var/did_anything = TRUE
 			switch(last_genital.type)
 				if(/obj/item/organ/genital/penis)
-					message = "cums into \the <b>[partner]</b>!"
+					message = "cums into \the <b>[partner_name]</b>!"
 				if(/obj/item/organ/genital/vagina)
-					message = "squirts into \the <b>[partner]</b>!"
+					message = "squirts into \the <b>[partner_name]</b>!"
 				else
 					did_anything = FALSE
 			if(did_anything)
 				LAZYADD(obscure_to, src)
 	else //todo: better self cum messages
 		message = "cums all over themselves!"
+
+	if(partner_carbon_check && cumin)
+		switch(target_orifice)
+			if(CUM_TARGET_VAGINA)
+				target_gen = c_partner.getorganslot(ORGAN_SLOT_VAGINA)
+			if(CUM_TARGET_ANUS)
+				target_gen = c_partner.getorganslot(ORGAN_SLOT_ANUS)
+			if(CUM_TARGET_BREASTS)
+				target_gen = c_partner.getorganslot(ORGAN_SLOT_BREASTS)
+			if(CUM_TARGET_PENIS)
+				target_gen = c_partner.getorganslot(ORGAN_SLOT_PENIS)
+
```

Comment:
I believe this is already handled on the interactions themselves

Reply:
> Only in 2 cases, and only for the female genital. And those 2 cases should be removed in favor of this because this is a better way of doing this, unless you disagree

I suppose that's fair, though remember it's a SPLURT addition so you might wanna check it also doesn't cause issues with the code already in place, also the override in modular_splurt